### PR TITLE
Another set of misc content fixes and changes.

### PR DESCRIPTION
**Pull request by hsaliak**
- Fixed quickstarts to remove redundant header.
- Updated docs section overview blurb to point to the right location
- Replaced generic text with information in the guides section and fixed up links.
- Wrote tutorials index and made reference to proto-lite, dropped reference to nanoproto.

These changes are more fix toward master tracking issue #258

### CODE REVIEW

File: docs/guides/index.md [hunk 1]
```diff
@@ -25,6 +29,6 @@ By default gRPC uses protocol buffers, Google’s mature open source mechanism f
...
+While protocol buffers have been available for open source users for some time, our examples use a new flavor of protocol buffers called proto3, which has a slightly simplified syntax, some useful new features, and supports lots more languages. This is currently available in Java, C++, Python, Objective-C, and C#, and a lite-runtime(Android Java), Ruby, and JavaScript from the protocol buffers Github repo, as well as a Go language generator from the golang/protobuf Github repo, with more languages in development. You can find out more in the proto3 language guide and the reference documentation for each language (where available), and see the major differences from the current default version in the release notes. Even more proto3 documentation is coming soon.
```
Comment (LisaFC) [inline docs/guides/index.md]: ...Objective-C, C#, a lite runtime for Android Java, Ruby, and JavaScript...

File: docs/tutorials/index.md [hunk 2]
```diff
@@ -7,48 +7,19 @@ type: markdown
...
-
-
-### [Section 3]
-[Body Copy] Lorem ipsum dolor sit amet, consectetur adipiscing elit. Sed blandit consequat finibus. Nullam ullamcorper ex sit amet laoreet mollis. Morbi quis porttitor sapein. Mauris malesuada libero a egestas iaculis.
-
-#### [Section 3.1]
-[Body Copy]Lorem ipsum dolor sit amet, consectetur adipiscing elit. Sed blandit consequat finibus. Nullam ullamcorper ex sit amet laoreet mollis. Morbi quis porttitor sapein. Mauris malesuada libero a egestas iaculis. Lorem ipsum dolor sit amet, consectetur adipiscing elit. Sed blandit consequat finibus. Nullam ullamcorper ex sit amet laoreet mollis. Morbi quis porttitor sapein. Mauris malesuada libero a egestas iaculis.
+These pages contain a collection of tutorials on the gRPC API and the kinds of RPCs that are available.
```
Comment (LisaFC) [inline docs/tutorials/index.md]: This section contains tutorials for each of our supported languages that introduce you to the gRPC API and associated concepts, and the different RPC types that are available. If you just want to dive straight in with a working example first, see our [Quickstarts](../quickstarts) - these tutorials are a little more in-depth.

File: docs/tutorials/index.md [hunk 3]
```diff
@@ -7,48 +7,19 @@ type: markdown
...
+ - [Java](basic/java.html)
+ - [Python](basic/python.html)
+ - [Go](basic/go.html)
+ - [Ruby](basic/ruby.html)
+ - [Node.js](basic/node.html)
+ - [C#](basic/csharp.html)
+ - [Objective-C](basic/objective-c.html)
+ - [PHP](basic/php.html)
+
+We also have a growing number of tutorials on a few specific topics.
```
Comment (LisaFC) [inline docs/tutorials/index.md]: We also have a growing number of tutorials on follow-on topics, with more in the pipeline.

File: docs/tutorials/index.md [hunk 4]
```diff
@@ -7,48 +7,19 @@ type: markdown
...
+
+- [Async APIs](async/helloasync-cpp.html)
```
Comment (LisaFC) [inline docs/tutorials/index.md]: Should we specify that this is in C++?

**Comment by LisaFC** [comment]
LGTM once a couple of small nits are resolved.

File: docs/tutorials/index.md (hunk 2 above)
Comment (hsaliak) [inline docs/tutorials/index.md]: Changed to:
This section contains tutorials for each of our supported languages. They introduce you to gRPC’s API and associated concepts, and the different RPC types that are available. If you just want to dive straight in with a working example first, see our Quickstarts.

File: docs/tutorials/index.md (hunk 4 above)
Comment (hsaliak) [inline docs/tutorials/index.md]: ack

**Comment by hsaliak** [comment]
@LisaFC this should address your review comments, PTAL

File: docs/quickstart/index.md [hunk 5]
```diff
@@ -8,40 +8,25 @@ sidenav: doc-side-quickstart-nav.html
...
 ---
 <p class="lead">
-All you need to get started with gRPC
+Get started with gRPC
 </p>
 
 <div id="toc" class="toc mobile-toc"></div>
 
-These pages show you how to get up and running as quickly as possible in gRPC, including installing all the tools you’ll need and completing some basic tasks to run and modify a simple Hello World application. There’s a Quick Start for each gRPC supported language with accompanying sample code for you to explore and update. For a more detailed look at the same application and how it all works, see [Hello gRPC!] in our Guides section. For more tutorials and examples, see our [Tutorials]. You can read more about gRPC in general in [What is gRPC?]. 
+These pages show you how to get up and running as quickly as possible in gRPC, including installing all the tools you’ll need.
```
Comment (LisaFC) [inline docs/quickstart/index.md]: Can we put the sentence you removed back in? It actually puts the Quickstarts in context. Though I've just realized that we're missing Hello gRPC! in its entirety - it's basically the other half of the original Overview.  I'll do a separate PR with it tomorrow.

File: docs/tutorials/index.md [hunk 6]
```diff
@@ -7,48 +7,19 @@ type: markdown
...
-
-
-### [Section 3]
-[Body Copy] Lorem ipsum dolor sit amet, consectetur adipiscing elit. Sed blandit consequat finibus. Nullam ullamcorper ex sit amet laoreet mollis. Morbi quis porttitor sapein. Mauris malesuada libero a egestas iaculis.
-
-#### [Section 3.1]
-[Body Copy]Lorem ipsum dolor sit amet, consectetur adipiscing elit. Sed blandit consequat finibus. Nullam ullamcorper ex sit amet laoreet mollis. Morbi quis porttitor sapein. Mauris malesuada libero a egestas iaculis. Lorem ipsum dolor sit amet, consectetur adipiscing elit. Sed blandit consequat finibus. Nullam ullamcorper ex sit amet laoreet mollis. Morbi quis porttitor sapein. Mauris malesuada libero a egestas iaculis.
+This section contains tutorials for each of our supported languages. They introduce you to gRPC's API and associated concepts, and the different RPC types that are available. If you just want to dive straight in with a working example first, see our [Quickstarts](../quickstart).
+
```
Comment (LisaFC) [inline docs/tutorials/index.md]: Just realized it's probably better to do a more qualified link, ie [Quickstarts]({{site.baseurl}}/docs/quickstart)

File: docs/tutorials/index.md [hunk 7]
```diff
@@ -7,48 +7,19 @@ type: markdown
...
+ - [Ruby](basic/ruby.html)
+ - [Node.js](basic/node.html)
+ - [C#](basic/csharp.html)
+ - [Objective-C](basic/objective-c.html)
+ - [PHP](basic/php.html)
+
+We also have a growing number of tutorials on follow-on topics, with more in the pipeline.
+
+- [Async APIs in C++](async/helloasync-cpp.html)
+- [Oauth2 on gRPC from Objective](auth/oauth2-objective-c.html)
```
Comment (LisaFC) [inline docs/tutorials/index.md]: from Objective C

**Comment by LisaFC** [comment]
One or two more minor nits, then LGTM.

File: docs/tutorials/index.md (hunk 6 above)
Comment (hsaliak) [inline docs/tutorials/index.md]: I have not been doing it so far. But I can switch this one around.

**Comment by hsaliak** [comment]
Done, ptal

File: docs/guides/index.md [hunk 8]
```diff
@@ -7,9 +7,13 @@ title: Guides
 type: markdown
 customjs: //survey.g.doubleclick.net/async_survey?site=dgxvheagfp4ai24o6y2ammm5fe
 ---
-<p class="lead">
-This document introduces you to gRPC (and its friend, protocol buffers).  If you’re new to gRPC and/or protocol buffers, read this! If you just want to dive in and see gRPC in action first, see our [Quick Starts].
-</p>
+
+This document introduces you to gRPC and protocol buffers. gRPC can use
+protocol buffers as both it's IDL and as its underlying message
```
Comment (LisaFC) [inline docs/guides/index.md]: its IDL

**Comment by LisaFC** [comment]
LGTM

**Pull request merged**